receiver/azureblobrehydrationreceiver/receiver.go

```diff
@@ -109,170 +120,185 @@ func newRehydrationReceiver(id component.ID, logger *zap.Logger, cfg *Config) (*
 		return nil, fmt.Errorf("invalid ending_time timestamp: %w", err)
 	}
 
-	ctx, cancel := context.WithCancelCause(context.Background())
-
 	return &rehydrationReceiver{
 		logger:          logger,
 		id:              id,
 		cfg:             cfg,
 		azureClient:     azureClient,
-		doneChan:        make(chan struct{}),
 		checkpointStore: rehydration.NewNopStorage(),
 		startingTime:    startingTime,
 		endingTime:      endingTime,
-		ctx:             ctx,
-		cancelFunc:      cancel,
+		blobChan:        make(chan *azureblob.BlobResults),
```

Comment:
I think we should consider making this a buffered channel. I think it would be okay for the receiver to slow to a crawl if the channel is full and we are waiting on blobs to be processed and sent down the pipeline.

Reply:
Good idea 👍